hazelcast-tpc-engine/src/test/java/com/hazelcast/internal/tpc/AsyncSocket_LargePayloadTest.java

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.tpc;
+
+import com.hazelcast.internal.tpc.iobuffer.IOBuffer;
+import com.hazelcast.internal.tpc.iobuffer.IOBufferAllocator;
+import com.hazelcast.internal.tpc.iobuffer.NonConcurrentIOBufferAllocator;
+import com.hazelcast.internal.tpc.util.IOUtil;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.nio.ByteBuffer;
+import java.util.concurrent.CountDownLatch;
+
+import static com.hazelcast.internal.tpc.TpcTestSupport.assertOpenEventually;
+import static com.hazelcast.internal.tpc.util.BitUtil.SIZEOF_INT;
+import static com.hazelcast.internal.tpc.util.BitUtil.SIZEOF_LONG;
+import static com.hazelcast.internal.tpc.util.BufferUtil.put;
+import static com.hazelcast.internal.tpc.util.IOUtil.closeResource;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public abstract class AsyncSocket_LargePayloadTest {
+    // use small buffers to cause a lot of network scheduling overhead (and shake down problems)
+    public static final int SOCKET_BUFFER_SIZE = 16 * 1024;
+    public static int requestTotal = 20000;
+
+    private Eventloop clientEventloop;
+    private Eventloop serverEventloop;
+    private AsyncSocket clientSocket;
+    private AsyncServerSocket serverSocket;
+
+    public abstract Eventloop createEventloop();
+
+    @Before
+    public void before() {
+        clientEventloop = createEventloop();
+        serverEventloop = createEventloop();
+    }
+
+    @After
+    public void after() throws InterruptedException {
+        closeResource(clientSocket);
+        closeResource(serverSocket);
+
+        if (clientEventloop != null) {
+            clientEventloop.shutdown();
+            clientEventloop.awaitTermination(10, SECONDS);
+        }
+
+        if (serverEventloop != null) {
+            serverEventloop.shutdown();
+            serverEventloop.awaitTermination(10, SECONDS);
+        }
+    }
+
+    public class Banana{}
```

Comment:
I was struggling with junit because it wanted a class to be public and have non static methods. I got baffled by this requirement even though the class isn't a test class.. until I figured out that instead of adding @Override, I added @Test :)